bftengine/include/bftengine/CryptoManager.hpp

```diff
@@ -23,7 +23,9 @@ namespace bftEngine {
 class CryptoManager : public IKeyExchanger, public IMultiSigKeyGenerator {
  public:
   static CryptoManager& instance(const ReplicaConfig* config = nullptr, Cryptosystem* cryptoSys = nullptr) {
-    static CryptoManager cm_(config, cryptoSys);
+    // ensure cryptoSys will be freed since we're expected to take ownership
+    std::unique_ptr<Cryptosystem> cryptoSysPtr(cryptoSys);
```

Comment:
I'm not sure why we need this intermediate `unique_ptr` here... In the `CryptoManager`, the raw pointer is reset into the `multiSigCryptoSystem_` unique_ptr. So I don't see how the raw pointer would leak, even without your changes?

Reply:
The CryptoManager construction is only called the first time. For subsequent calls the raw pointer is left dangling.